examples/performance-subscribe/src/PerformanceTestApp.cpp

```diff
@@ -0,0 +1,121 @@
+                         const std::chrono::high_resolution_clock::time_point timestamp) {
+    std::time_t time = std::chrono::high_resolution_clock::to_time_t(timestamp);
+    uint64_t    useconds =
+        std::chrono::duration_cast<std::chrono::microseconds>(timestamp.time_since_epoch())
+            .count() %
+        std::micro::den;
+    ostr << std::put_time(std::localtime(&time), "%T") << "." << std::setw(6) << std::setfill('0')
+         << useconds;
+    return ostr;
+}
+
+std::vector<std::string> readSignalNames(const std::string& configFile) {
+    velocitas::logger().info("Reading signal list from file {}.", configFile);
+    auto config     = nlohmann::json::parse(std::ifstream(configFile));
+    auto signalList = config["signals"];
+
+    std::vector<std::string> signalNames;
+    signalNames.reserve(signalList.size());
+    for (const auto& signal : signalList) {
+        const auto& signalName = signal["path"];
+        if (!signalName.empty()) {
+            velocitas::logger().debug("    {}", signalName);
+            signalNames.push_back(signalName);
+        }
+    }
+    return signalNames;
+}
+
+} // anonymous namespace
+
+PerformanceTestApp::PerformanceTestApp(std::string configFile)
+    : VehicleApp(velocitas::IVehicleDataBrokerClient::createInstance("vehicledatabroker"),
+                 velocitas::IPubSubClient::createInstance("PerformanceTestApp"))
+    , m_configFile{std::move(configFile)} {}
+
+void PerformanceTestApp::onStart() {
+    auto dataPointList = readSignalNames(m_configFile);
+    velocitas::logger().info("Subscribing to signals ...");
+    for (auto path : dataPointList) {
+        subscribeDataPoints("SELECT " + path)
+            ->onItem([path](const velocitas::DataPointReply& reply) {
+                auto value     = getValueRepresentation(*reply.getGeneric(path));
+                auto timestamp = std::chrono::high_resolution_clock::now();
+                std::cout << timestamp << " - " << path << " - " << value << std::endl;
+            })
+            ->onError([path](const velocitas::Status& status) {
+                velocitas::logger().error("Error on subscription for data point {}: {}", path,
+                                          status.errorMessage());
+            });
+    }
+}
+
+} // namespace example
+
+namespace {
+
+std::unique_ptr<example::PerformanceTestApp> myApp;
+
+void signal_handler(int sig) {
+    velocitas::logger().info("App terminating signal received: {}", sig);
+    myApp->stop();
+}
+
+const char DEFAULT_CONFIG_FILENAME[] = "subscription_signals.json";
+
+std::string getDefaultConfigFilePath(const char* appBinaryPath) {
+    std::filesystem::path path = appBinaryPath;
+    path.replace_filename(DEFAULT_CONFIG_FILENAME);
+    return std::string{path};
+}
+
+} // namespace
+
+int main(int argc, char** argv) {
+    signal(SIGINT, signal_handler);
+
+    const auto configFile = (argc > 1) ? std::string(argv[1]) : getDefaultConfigFilePath(argv[0]);
```

Comment:
Reading the signal file should be done before creating the app as it is a necessary pre-condition for the file to exist. I suggest we only pass the created vector of signal names to the PerformanceTestApp constructor